pallets/identity/src/lib.rs

```diff
@@ -583,7 +582,8 @@ decl_module! {
                 Ok(post_info) => post_info.actual_weight,
                 Err(err) => err.post_info.actual_weight,
             };
-            Ok((Some(actual_weight.unwrap_or(0u64) + 500_000 + proposal_weight as Weight)).into())
+
+            Ok((actual_weight.map(|w| w + 500_000)).into())
```

Comment:
```suggestion
            Ok((actual_weight.unwrap_or(0) + 500_000).into())
```
We want a min weight to be applicable here even if the internal call was weight-free for some reason.

Reply:
Yeah so if the internal call says `None` as the `actual_weight` then it will return the None otherwise `Some(actual_weight + 500_000)` that will always be < `call.get_dispatch_info().weight + 500_000`

Reply:
I am concerned about the None case here. The actual weight of this wrapper transaction should never be none. It should be 500k + weight of inner_transaction.

Reply:
Initially, I did also think the same but it will work something like this - 

If `None` comes from the internal transaction then it means whatever the weight is used from the proposal is right and actual weight to use. So for the upstream txn there is nothing to refund as everything that is deducted upfront was the right figure. and return the `None` as the `actual_weight` for the post_dispatch info.

Even If I return the 500k + weight of the inner transaction behaviour will be same as `calc_actual_weight` picks up the minimum between the weight used in the upfront and the actual_weight present in the post_info.

Reply:
>If None comes from the internal transaction then it means whatever the weight is used from the proposal is right and actual weight to use.

So if the actual weight is None, that means the actual weight is the max weight. This sounds like a weird API choice from Parity. Can you please test it to double-check?

If this is indeed the case, then we don't need to make any changes to the code but please add a dev comment about it. It's a very non-intuitive thing imo.

Reply:
Yeah it worked like that either you have to send the actual_weight that should be less than or equal to the upfront maximum weight or you can send `None` which means Max upfront weight is the right weight to use and refunds are 0.